File: lab-1/domain/album.py
from domain.media_item import media_item


class Album(media_item):
    def __init__(self, title, creation_date, artist, genre, no_songs):
        super().__init__(title, creation_date)
        self.artist = artist
        self.genre = genre
        self.no_songs = no_songs

    def play(self):
        print(f"Playing the album: {self.title} by {self.artist}")

    def display_info(self):
        print(f"Album: {self.title} by {self.artist}")
        print(f"Genre: {self.genre}")
        print(f"Release date: {self.creation_date}")
        print(f"Number of songs: {self.no_songs}")

File: src/main/kotlin/media_item/implementations/Album.kt
package media_item.implementations
import media_item.MediaItem
import javax.print.attribute.standard.Media

class Album(
        override val title: String,
        override val creationDate: String,
        val artist: String,
        val genre: String,
        val noSongs: Int
) : MediaItem() {
    val items = mutableListOf<MediaItem>()

    override fun play() {
        println("Playing the album: $title by $artist")
    }

    override fun displayInfo() {
        println("Album: $title by $artist")
        println("Genre: $genre")
        println("Release date: $creationDate")
        println("Number of songs: $noSongs")
        println("Items:")
        items.forEach {
            println(it.title)
        }
    }

    override fun clone(): Album {
        return Album(title, creationDate, artist, genre, noSongs)
    }

    override fun toString(): String {
        return "Album(title='$title', creationDate='$creationDate', artist='$artist', genre='$genre', noSongs=$noSongs)"
    }
}


